File: Dart/uno/lib/twoSum.dart
class twoSum {
    List <int> twoSum(List <int> nums, int target) {
        List <int> result = [];
        for (int i = 0; i < nums.length; i++) {
            for (int j = i+1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    result.add(i);
                    result.add(j);
                    return result;
                }
            }
        }
        return result;
    }
    int calculate() {
      // ~/ is integer division
      return 6 * 7 ~/ 2;
    }
    List<int> twoSum(List<int> nums, int target) {
      final map = {};
      for(int i= 0;i < nums.length; i++) {
        final match = target - nums[i];
        if(map.containsKey(match)){
          return [i, map[match]];
        }
        map[nums[i]] = i;
      }
      return [];
    }
}

File: csharp/twoSum/twoSum.cs
namespace Csharp.TwoSum
{
    public class TwoSum
    {
        public int[] twoSum(int[] nums, int target)
        {
            int[] result = new int[2];
            for (int i = 0; i < nums.Length; i++)
            {
                for (int j = i + 1; j < nums.Length; j++)
                {
                    if (nums[i] + nums[j] == target)
                    {
                        result[0] = i;
                        result[1] = j;
                    }
                }
            }
            return result;
        }
        public int[] TwoSumRobada(int[] nums, int target)
        {
            Dictionary<int, int> numDict = new Dictionary<int, int>();

            for (int i = 0; i < nums.Length; i++)
            {
                var complement = target - nums[i];

                if (numDict.ContainsKey(complement))
                {
                    var compValue = numDict[complement];
                    return new int[] { compValue, i };
                }
                else if (!numDict.ContainsKey(nums[i]))
                {
                    numDict.Add(nums[i], i);
                }
            }
            return null;
        }
    }
}
